### PR TITLE
add flag so translation can be locked while rotating: in order to inc…

### PR DESCRIPTION
**Pull request by jamesashley1**
…rease precision and eliminate jitteriness.

## Overview

Manipulating rotation, translation and scale at the same time can be awkward. To some extent scale and rotation can be disabled. This property allows translation to be locked under certain circumstances by the developer (for instance, lock down translation during rotation).

### CODE REVIEW

**Comment by SimonDarksideJ** [comment]
Thanks @jamesashley1 do you have an example or sample to demonstrate the use of this?

File: XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs [hunk 1]
```diff
@@ -396,6 +396,11 @@ public Vector2 NudgeConstraints
         /// </summary>
         public bool IsRotating => !updatedAngle.Equals(0f);
 
+        /// <summary>Gets or sets a value indicating whether this instance can be moved in space.</summary>
+        /// <value>
+        ///   <c>true</c> if this instance is translate locked; otherwise, <c>false</c>.</value>
+        public bool IsTranslateLocked { get; set; } = false;
```
Comment (StephenHodgson) [inline XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs]: Should we add a serialized field here to configure it in the inspector?

File: XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs [hunk 2]
```diff
@@ -396,6 +396,11 @@ public Vector2 NudgeConstraints
         /// </summary>
         public bool IsRotating => !updatedAngle.Equals(0f);
 
+        /// <summary>Gets or sets a value indicating whether this instance can be moved in space.</summary>
+        /// <value>
+        ///   <c>true</c> if this instance is translate locked; otherwise, <c>false</c>.</value>
```
Comment (StephenHodgson) [inline XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs]: The formatting on this seems off

**Comment by jamesashley1** [comment]
@SimonDarksideJ I'm using it in a derived class to dynamically lock the translation with a hand gesture:

    public override void ProcessTransformData()
    {
        if (!IsBeingHeld || PrimaryPointer == null) { return; }
        IsTranslateLocked = IsStopHandGestureActive();
        ...

File: XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs (hunk 2 above)
Comment (jamesashley1) [reply XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs]: fixed, I believe.

File: XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs (hunk 1 above)
Comment (jamesashley1) [reply XRTK.SDK/Packages/com.xrtk.sdk/Features/Input/Handlers/ManipulationHandler.cs]: fixed.

**Comment by StephenHodgson** [comment]
When trying this out I noticed that the position would drift as I moved the controller and rotated.

I'm wonder if there's something else we're missing to make sure the position and rotation is being correctly calculated when the position is locked.

**Comment by StephenHodgson** [comment]
Previous comment was testing out something in our app, but it doesn't seem to be drifting anymore so it might have been something there.

**Comment by StephenHodgson** [comment]
Guys we really need to be more aware of the branch we're targeting with these pull requests

**Pull request merged**